File: src/model/Partida.java
package model;

import java.time.LocalDate;
import java.util.ArrayList;

public class Partida {
    private String resultado;
    private String adversario;
    private int golsPro;
    private int golsContra;
    private LocalDate data;
    private ArrayList<Atleta> escalacao;
    private Atleta mvp;

    public Partida(int golsPro, int golsContra) {
        this.golsPro = golsPro;
        this.golsContra = golsContra;
        calcRes();
        this.escalacao = new ArrayList<>();
    }

    public String getResultado() {
        return resultado;
    }

    public void setResultado(String resultado) {
        this.resultado = resultado;
    }

    public String getAdversario() {
        return adversario;
    }

    public void setAdversario(String adversario) {
        this.adversario = adversario;
    }

    public int getGolsPro() {
        return golsPro;
    }

    public void setGolsPro(int golsPro) {
        this.golsPro = golsPro;
    }

    public int getGolsContra() {
        return golsContra;
    }

    public void setGolsContra(int golsContra) {
        this.golsContra = golsContra;
    }

    public LocalDate getData() {
        return data;
    }

    public void setData(LocalDate data) {
        this.data = data;
    }

    public ArrayList<Atleta> getEscalacao() {
        return escalacao;
    }

   /* public void setEscalacao(ArrayList<Atleta> escalacao) {
        this.escalacao = escalacao;
    }*/

    public Atleta getMvp() {
        return mvp;
    }

    public void setMvp(Atleta mvp) {
        this.mvp = mvp;
    }

    public void addEscalacao(Atleta atleta) {
        this.escalacao.add(atleta);
    }

    public String calcRes() {
        if (golsPro > golsContra) {
            resultado = "Vitoria";
        } else if (golsPro == golsContra) {
            resultado = "Empate";
        }else {
            resultado = "Derrota";
        }
        return resultado;
    }

    @Override
    public String toString() {
        return "Adversario: " + adversario + ", Resultado: " + resultado;
    }
}

File: sgfc_too/lib/Partida.dart
import '../atleta.dart';

class Partida {
  String _resultado = '';
  String _adversario = '';
  int _golsPro;
  int _golsContra;
  DateTime? _data;
  final List<Atleta> _escalacao = [];
  Atleta? _mvp;

  Partida({
    required int golsPro,
    required int golsContra,
  })  : _golsPro = golsPro,
        _golsContra = golsContra {
    _calcRes();
  }

  // Getters e Setters
  String get resultado => _resultado;
  set resultado(String value) => _resultado = value;

  String get adversario => _adversario;
  set adversario(String value) => _adversario = value;

  int get golsPro => _golsPro;
  set golsPro(int value) => _golsPro = value;

  int get golsContra => _golsContra;
  set golsContra(int value) => _golsContra = value;

  DateTime? get data => _data;
  set data(DateTime? value) => _data = value;

  List<Atleta> get escalacao => _escalacao;

  Atleta? get mvp => _mvp;
  set mvp(Atleta? value) => _mvp = value;

  void addEscalacao(Atleta atleta) {
    _escalacao.add(atleta);
  }

  String _calcRes() {
    if (_golsPro > _golsContra) {
      _resultado = 'Vitória';
    } else if (_golsPro == _golsContra) {
      _resultado = 'Empate';
    } else {
      _resultado = 'Derrota';
    }
    return _resultado;
  }

  @override
  String toString() {
    return 'Adversário: $_adversario, Resultado: $_resultado';
  }
}
